File: src/main/java/org/cascadebot/cascadebot/data/objects/donation/DataFlag.java
package org.cascadebot.cascadebot.data.objects.donation;

import com.google.gson.JsonObject;
import org.cascadebot.cascadebot.data.language.Locale;

public abstract class DataFlag extends Flag {

    public DataFlag(String id, FlagScope scope) {
        super(id, scope);
    }

    protected DataFlag() {
        super();
    }

    abstract DataFlag parseFlagData(JsonObject flagDataObject);

    abstract public String toString();

    abstract public String getDescription(Locale locale);

}

File: src/main/kotlin/org/cascadebot/cascadebot/data/objects/donation/DataFlag.kt
package org.cascadebot.cascadebot.data.objects.donation

import com.google.gson.JsonObject
import org.cascadebot.cascadebot.data.language.Locale

abstract class DataFlag : Flag {
    constructor(id: String, scope: FlagScope) : super(id, scope)
    protected constructor() : super()

    abstract fun parseFlagData(flagDataObject: JsonObject): DataFlag
    abstract override fun getDescription(locale: Locale): String

    abstract operator fun compareTo(flag: DataFlag): Int
}